src/bit-fields-macros/src/builder.rs

```diff
@@ -0,0 +1,816 @@
+        #[cfg(feature = "field_map")]
+        let (field_matching_from_hashmap, field_setting_hashmap) =
+            (self.field_matching_from_hashmap, self.field_setting_hashmap);
+        #[cfg(feature = "set_theory")]
+        let (
+            fields_superset_fn,
+            fields_subset_fn,
+            fields_disjoint_fn,
+            fields_intersection_fn,
+            fields_union_fn,
+        ) = (
+            self.fields_superset_fn,
+            self.fields_subset_fn,
+            self.fields_disjoint_fn,
+            self.fields_intersection_fn,
+            self.fields_union_fn,
+        );
+        #[cfg(feature = "display")]
+        let display_fmt_string = self.display_fmt_string;
+        let struct_doc_table_layout: TokenStream = self
+            .struct_doc_table_layout
+            .parse()
+            .expect("Failed to parse");
+        let struct_accessors = self.struct_accessors;
+
+        let struct_name = self.struct_name;
+        let struct_data_type = self.data_type;
+        let equal_defined_bits = self.equal_defined_bits;
+
+        #[cfg(feature = "serde")]
+        let visitor = quote::format_ident!("{}Visitor", struct_name);
+
+        #[cfg(feature = "serde")]
+        let serde = quote! {
+            impl serde::Serialize for #struct_name {
+                fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                where
+                    S: serde::Serializer,
+                {
+                    use serde::ser::{Serialize, SerializeMap, SerializeSeq, SerializeTuple, Serializer};
+                    let (set, map): (std::collections::HashSet<String>, std::collections::HashMap<String, #struct_data_type>) = self.into();
+                    let mut tup = serializer.serialize_tuple(2)?;
+                    tup.serialize_element(&set)?;
+                    tup.serialize_element(&map)?;
+                    tup.end()
+                }
+            }
+            struct #visitor;
+            impl<'de> serde::de::Visitor<'de> for #visitor {
+                type Value = #struct_name;
+                fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                    write!(
+                        formatter,
+                        "a set of feature flags followed by a map of fields"
+                    )
+                }
+                fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+                where
+                    A: serde::de::SeqAccess<'de>,
+                {
+                    use std::convert::TryFrom;
+                    if let Some(set) = seq.next_element::<std::collections::HashSet<String>>()? {
+                        if let Some(map) = seq.next_element::<std::collections::HashMap<String, #struct_data_type>>()? {
+                            Ok(#struct_name::try_from((set, map)).expect("Failed to deserialize #struct_name"))
```

Comment:
The final `#struct-name` is not interpolated, as its inside of a string

Reply:
Yep you're right, I forgot this. Fixed.